### PR TITLE
Enable declaring sharing info in JSON

### PR DESCRIPTION
**Pull request by SethPoulsen**
This PR is the first in a series of a few that will implement #10005. 

Here, we allow sharing info to be present in the JSON, however we do not yet do anything with this information.
We will deploy this first in order to allow beta sharing users the opportunity to add sharing info to their JSON, before deploying the changes that will make their sync error if they _don't_ have the sharing info in their JSON.

### CODE REVIEW

File: apps/prairielearn/src/schemas/schemas/infoQuestion.json
Comment (SethPoulsen) [inline apps/prairielearn/src/schemas/schemas/infoQuestion.json:269]: do we want this name to be shorter in the JSON than the actual name in the DB? or are we happy with this? seems fine to me

File: apps/prairielearn/src/lib/editors.ts
Comment (SethPoulsen) [inline apps/prairielearn/src/lib/editors.ts:1059]: I was on the fence about deleting sharing info when copying a question within a course, but the fact that sharing can't be undone at this point makes me think we should not share by default when copying.

File: apps/prairielearn/src/lib/editors.ts
Comment (SethPoulsen) [inline apps/prairielearn/src/lib/editors.ts:1133]: this should solve our problem nicely for people copying public questions into their own course

File: apps/prairielearn/src/schemas/schemas/infoQuestion.json
Comment (nwalters512) [inline apps/prairielearn/src/schemas/schemas/infoQuestion.json:1]: Biekshedding: let's consider putting all this in an object under a `sharing` key or something like that? Similarly to how we scope external grading and workspace config. Rationale: we're going to want to programmatically modify this from the UI, and it'd be nice if any related properties would automatically be visually grouped in the JSON file. Let's discuss before making any changes.

File: testCourse/questions/addNumbers/info.json
Comment (nwalters512) [inline testCourse/questions/addNumbers/info.json:11]: Nit: `sharedPublicly` / `sharedPubliclyWithSource`? This would match the DB columns.

File: apps/prairielearn/src/schemas/schemas/infoCourse.json
Comment (SethPoulsen) [reply apps/prairielearn/src/schemas/schemas/infoCourse.json:173]: fixed

File: apps/prairielearn/src/lib/editors.ts
Comment (SethPoulsen) [reply apps/prairielearn/src/lib/editors.ts:1059]: done

File: testCourse/questions/addNumbers/info.json
Comment (SethPoulsen) [reply testCourse/questions/addNumbers/info.json:11]: done, thanks

File: apps/prairielearn/src/schemas/schemas/infoQuestion.json
Comment (SethPoulsen) [reply apps/prairielearn/src/schemas/schemas/infoQuestion.json:1]: in favor of them being top level/separate: people won't have sharingsets on a question if they have public sharing. So it seems like overkill to have a parent property for it when it will usually only have one or at most two children

Comment (mwest1066) [reply apps/prairielearn/src/schemas/schemas/infoQuestion.json:1]: I agree that in general I like having things grouped together under a top-level key. My only hesitation in this case is that in practice we'll typically only ever have one of the three properties set. (Assuming that `sharedPubliclyWithSource` implies `sharedPublicly`?) There might be reasons why you'd have `sharingSets` and `sharedPublicly` (maybe you used to share it privately, and then made it public) but I think this is probably not the common case. I think this is what distinguishes this from the cases like `workspaceOptions`.

But it's probably clearer to have everything grouped together, even if there is typically only one key set. @nwalters512 want to make a final call here, with the extra context that there is typically only one key in use?

Comment (SethPoulsen) [reply apps/prairielearn/src/schemas/schemas/infoQuestion.json:1]: I had not thought about having sharedPubliclyWithSource implies sharedPublicly automatically, but ya we should probably do that. In that case, then they will definitely only need one of the three properties.

Comment (nwalters512) [reply apps/prairielearn/src/schemas/schemas/infoQuestion.json:1]: Ahh, that's a good point. In fact, we could probably add some sync warnings to help guide people ("sharing sets" is unnecessary with both "shared publicly" and "shared publicly with source", and "shared publicly" is unnecessary with "shared publicly with source").

Final call: let's leave things as top level properties.

Comment (SethPoulsen) [reply apps/prairielearn/src/schemas/schemas/infoQuestion.json:1]: Sounds good. 

Is this ready to merge then?

Comment (nwalters512) [reply apps/prairielearn/src/schemas/schemas/infoQuestion.json:1]: I'll do a final review on Monday and then we can land this!

File: apps/prairielearn/src/schemas/schemas/infoCourse.json
Comment (nwalters512) [inline apps/prairielearn/src/schemas/schemas/infoCourse.json:179]: Checking my understanding: this would be unused in the current DB schema, correct? Are there plans to support this soon?

File: testCourse/questions/addNumbers/info.json
Comment (nwalters512) [inline testCourse/questions/addNumbers/info.json:1]: From experience, tacking on random "extra" behaviors to things in the test course can end in confusion and brittle tests.

- If you want to lay the groundwork for a test that shares questions from the test course to the example course, create brand-new questions that can be used to test each of the new properties.
- If you just want to assert that questions with these properties sync without error until you land full syncing support, add a test to `src/tests/sync/questionsSync.test.ts`.

**Review by nwalters512 — APPROVED** [review]
If you're itching to get this in I won't block on the test course questions comment, but I will ask that you fix that in a fast-follow PR.

File: testCourse/questions/addNumbers/info.json
Comment (SethPoulsen) [reply testCourse/questions/addNumbers/info.json:1]: Ya, the plan is to handling the tests properly in the next PR, where I am actually syncing sharing info from JSON and then testing that the sharing worked.

File: apps/prairielearn/src/schemas/schemas/infoCourse.json
Comment (SethPoulsen) [reply apps/prairielearn/src/schemas/schemas/infoCourse.json:179]: that's correct, its not used in the DB right now. I just put it there because Matt and I figured we would want it at some point, but we can change things if we want.

**Pull request merged**
